app/javascript/packs/building_map.js

```diff
@@ -120,3 +129,44 @@ function requestLocation(inputField){
         }
     );
 }
+
+const syncUserPositionWithServerImpl = async (location) => {
+    const body = new URLSearchParams({ location });
+    const response = await fetch(
+        "/users/geo_location",
+        {
+            method: "PUT",
+            body,
+            headers: {
+                "X-CSRF-TOKEN": csrfToken(),
+            },
+        }
+    );
+    console.assert(response.status === 204); // HTTP "No content"
+};
+const syncUserPositionWithServer = ratelimit(syncUserPositionWithServerImpl, 10000);
+
+let watcher_id;
+const positionIcon = L.icon({ iconUrl: "assets/current-location.svg", iconSize: [30, 30], iconAnchor: [15, 15] });
+const positionMarker = L.marker([0, 0], { icon: positionIcon });
+positionMarker.bindPopup("Your position");
+
+function trackingHandler() {
+    const tracking_switch = document.getElementById("tracking_switch");
```

Comment:
We should also rename `tracking_switch` and `watcher_id` to Camel Case to be consistent (and also do this in the HTML).